prov/gni/src/gnix_msg.c

```diff
@@ -421,9 +584,12 @@ static int __gnix_rndzv_req(void *arg)
 	int head_off, head_len, tail_len;
 	void *tail_data = NULL;
 
-	if (!req->msg.recv_md) {
+	GNIX_TRACE(FI_LOG_EP_DATA, "\n");
+
+	if (!req->msg.recv_md[0]) {
 		rc = gnix_mr_reg(&ep->domain->domain_fid.fid,
-				 (void *)req->msg.recv_addr, req->msg.recv_len,
+				 (void *)req->msg.recv_info[0].recv_addr,
+				 req->msg.recv_info[0].recv_len,
 				 FI_READ | FI_WRITE, 0, 0, 0, &auto_mr, NULL);
 		if (rc != FI_SUCCESS) {
 			GNIX_INFO(FI_LOG_EP_DATA,
```

Comment:
`gnix_mr_reg` was changed to *not* warn on failure, leaving it up to the callers.  Because of that, I think this (and similar call sites) should be changed to GNIX_WARN.